test/upgrade/dry_run_does_not_apply_changes_test.dart

```diff
@@ -50,6 +50,50 @@ void main() {
     ]).validate();
   });
 
+  test('--dry-run: shows count of discontinued packages', () async {
+    await servePackages((builder) {
+      builder.serve('foo', '2.0.0');
+    });
+
+    globalPackageServer.add((builder) => builder..discontinue('foo'));
+
+    // Create the first lockfile.
+    await d.appDir({'foo': '2.0.0'}).create();
+
+    await pubGet();
+
+    await d.dir(appPath, [
+      d.file('pubspec.lock', contains('2.0.0')),
+      d.dir('.dart_tool'),
+    ]).validate();
+
+    // Also delete the ".dart_tool" directory.
+    deleteEntry(path.join(d.sandbox, appPath, '.dart_tool'));
+
+    // Do the dry run.
+    await pubUpgrade(
+      args: ['--dry-run'],
+      output: contains('1 package is discontinued.'),
+    );
+
+    await d.dir(appPath, [
+      // The lockfile should be unmodified.
+      d.file('pubspec.lock', contains('2.0.0')),
+      // The ".dart_tool" directory should not have been regenerated.
+      d.nothing('.dart_tool')
+    ]).validate();
```

Comment:
I don't think we need this validation in this test. We want to test that we print the count, not how --dry-run behaves.